exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SQLExpressionBuilder.kt

```diff
@@ -396,7 +396,10 @@ interface ISqlExpressionBuilder {
     // Comparison Predicates
 
     /** Returns `true` if this expression is between the values [from] and [to], `false` otherwise. */
-    fun <T, S : T?> ExpressionWithColumnType<S>.between(from: T, to: T): Between = Between(this, wrap(from), wrap(to))
+    fun <T, S : T?> ExpressionWithColumnType<in S>.between(from: T, to: T): Between = Between(this, wrap(from), wrap(to))
+
+    /** Returns `true` if this [EntityID] expression is between the values [from] and [to], `false` otherwise. */
+    fun <T : Comparable<T>, E : EntityID<T>?> ExpressionWithColumnType<E>.between(from: T, to: T): Between = Between(this, wrap(from), wrap(to))
```

Comment:
As mentioned above, I'm not sure how to test for the presence of a compiler warning as all existing tests use arguments with the correct type. And any test purposefully set up to show a warning would fail when we upgrade the Kotlin version.

Please let me know if there's any type of test that should be added.

Reply:
It's fine not to test it